### ISSUE TITLE
feat(cdk/drag-drop): X and Y as percentage values instead of px

### ISSUE THREAD

**Issue by SvenBudak** [angular/components#28607]
### Feature Description

We are developing a mobile app in which users should be able to move elements back and forth in a certain area. The position of the elements should obviously be identical for every user. However, this does not work since each user has a different smartphone and therefore a different screen width, as cdk/drag-drop works with px values. One should be able to switch from px to % in order to ensure the correct position regardless of the screen resolution (or the width of the boundary element).

**Comment by crisbeto** [comment]
This may be tricky to do, because we currently map 1px of movement to 1px in the `transform` value.

**Comment by SvenBudak** [comment]
> This may be tricky to do, because we currently map 1px of movement to 1px in the `transform` value.

My math skills are at the level of a 12-year-old. 😅 Otherwise, I would participate in solving this problem. But, as I understand it, one must also scale the object to be moved by x% proportionally to the boundary. This ensures that the movable objects appear visually the same size at each scale.

I just realized that using transform: translate3d(97%, 100%, 0px) in percentage terms does not behave as I expected...

Unfortunately, this is really a problem, thus drag is not usable if it represents an element that is synchronized across multiple users, each with different screen widths.

**Comment by Lubjan** [comment]
I'm currently facing a similar problem where I have a container (that holds cdkDrag elements) which I scale using CSS transformation, if not mistaken the behaviour is similar to the one aforementioned here.
I think adding an Input like `cdkDragScaling` and incorporating it into the following could help. https://github.com/angular/components/blob/436301f41f0a44f60c99bda4d121f893a6b63d55/src/cdk/drag-drop/drag-ref.ts#L648
When using cdkDrag in an application, one could then for example set `[cdkDragScaling]="{ x: .75, y: .5 }"` and the scaling would be applied onto the elements movement. That would leave the work of calculating the scaling on the side of the dev using cdkDrag and would leave a lot of freedom, if I'm not entirely mistaken.

**Comment by Lubjan** [comment]
> @Lubjan Do I understand you correctly? The necessary features already exist in cdkDrag for clean scaling?

No, it is a proposal to add cdkDragScaling as a feature into cdkDrag itself. If something like that already exists, I am not aware of it or wasn't able to find it yet. (I don't mean visibly scaling the element)

**Comment by Lubjan** [comment]
It seems that a few months ago `cdkDragScale` was added to https://material.angular.io/cdk/drag-drop/api#DragRef.
Still no positions in percentage, but might solve most problems.

Relevant pr: [feat(cdk/drag-drop): add input to specify dragged item scale](https://github.com/angular/components/pull/29392)

**Comment by abdurrahmanyildirim** [comment]
To solve this issue, I introduced utility functions that convert pixel values to percentages relative to the container element. This allows the drag position to remain responsive to container size changes.

When applying positions via `cdkDragFreeDragPosition`, I use a pipe to convert those percentage values back into pixels, since the CDK API expects pixel-based coordinates.

A cleaner alternative might be to treat percentages as the primary input and rely on a `cdkDragBoundary` element that represents those percentage-based constraints. This would reduce the need for back-and-forth conversions and make the positioning logic more predictable and responsive.
